### ISSUE TITLE
Hillshade not created when Save File As is used

### ISSUE THREAD

**Issue by lauriefitzherbert** [qgis/QGIS#57247]
### What is the bug or the crash?

Raster Terrain Analysis module "Hillshade" produces a blank layer when the SAVE FILE AS... option that appears in the Hillshade dialog screen is used. 

But, the Hillshade module outputs a correct tif file layer when the above "SAVE FILE AS... " option is NOT USED.

PS
While taking time to write this "bug" I inadvertently discovered how this error occurs. (see Steps).  I'm a little embarrassed to post but hopeful my error could be made preventable thereby making a better experience for new users to get through the training manual.

### Steps to reproduce the issue

STEPS TO PRODUCE Blank Hillshade Layer:

1. In New Project, Layer->Add Layer->Add Raster layer-> Select File... 
( ie. /.. /raster/SRTM/srtm_41_19 obtained from QGIS Training Data Master zip )

2. Under Processing Toolbox->Raster terrain analysis->Hillshade 
(Settings used :  Z factor 1.0, Azimuth 300.0°, Vertical angle 40.0°; )

3. Right click on Hillshade text box, Select-> Save File As....
choose /directory/hillshade (or anyname or anyname.vrt)

4. Select ->Run. 
5. Close dialog.
RESULT: New Hillshade layer appears layer menu (as top layer) but layer is BLANK.  
Turning other layer off made no difference. Turning Hillshade layer on and off did nothing. Screen remained white.

STEPS THAT PRODUCED *CORRECT HILLSHADE LAYER
Follow steps 1 & 2 above.  Skip 3- DO NOT USE "Save File As", Leave Blank. Select Run and Close dialog
*matches appearance of QGIS Tutorial Lesson 7.3.1 version 3.34.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.36.1-Maidenhead | QGIS code revision | 3e589453264
-- | -- | -- | --
Qt version | 5.15.8
Python version | 3.11.2
GDAL/OGR version | 3.6.2
PROJ version | 9.1.1
EPSG Registry database version | v10.076 (2022-08-31)
GEOS version | 3.11.1-CAPI-1.17.1
SQLite version | 3.40.1
PostgreSQL client version | 15.5 (Debian 15.5-0+deb12u1)
SpatiaLite version | 5.0.1
QWT version | 6.1.4
QScintilla2 version | 2.13.3
OS version | Debian GNU/Linux 12 (bookworm)
  |   |   |  
Active Python plugins
valuetool | 3.0.19
grassprovider | 2.12.99
MetaSearch | 0.3.6
processing | 2.12.99
db_manager | 0.1.20

</body></html>QGIS version
3.36.1-Maidenhead
QGIS code revision
[3e589453264](https://github.com/qgis/QGIS/commit/3e589453264)
Qt version
5.15.8
Python version
3.11.2
GDAL/OGR version
3.6.2
PROJ version
9.1.1
EPSG Registry database version
v10.076 (2022-08-31)
GEOS version
3.11.1-CAPI-1.17.1
SQLite version
3.40.1
PostgreSQL client version
15.5 (Debian 15.5-0+deb12u1)
SpatiaLite version
5.0.1
QWT version
6.1.4
QScintilla2 version
2.13.3
OS version
Debian GNU/Linux 12 (bookworm)

Active Python plugins
valuetool
3.0.19
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99
db_manager
0.1.20

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

NOTE:

Following the QGIS Training Manual 3.34, chapt 7.3. But  I'd saved the file using the hillshade dialog.  I did not know that the  SAVE AS FILE, set to type "vrt". was wrong, nor did I have info on the file format required by the hillshade algorithm.   It was only after using export on an unsaved correctly displayed hillshade layer produced, that saved in tif format and finding this tif file imported correctly that I decided to investigate what file type I'd been saving. (vrt). At this point  I just happened to see other file format options in the hillshade dialog.  

PROBABLY, THERE IS GOOD REASON- 
but my question is: 
Why would one want a vrt file output while making a hillshade without producing a tiff file for the vrt to point to?

**Comment by elpaso** [comment]
@lauriefitzherbert I'm not sure is there anything we can do about this: if I understand it correctly, VRT is a valid option for saving the algorithm result.

**Comment by alexbruy** [comment]
I'd say this is a won't fix.

As already explained in the linked documentation ticket this only happens if last used format is VRT and user don't choose another output format. VRT is a valid output option for some raster algorithms and it is users responsibility to select output format.
